pkg/controller/acmeorders/sync.go

```diff
@@ -163,8 +186,34 @@ func (c *controller) Sync(ctx context.Context, o *cmacme.Order) (err error) {
 			}
 		}
 		return err
+
 	// anyChallengesFailed(challenges) == false is already implied by the above
-	// case, but explicitly check it here in case anything changes in future.
+	// case, but explicitly check it in the following cases for if anything changes in future.
+
+	// This is to avoid stuck Orders in edge cases where all the Challenges have
+	// been finalized, but the ACME server has not yet updated the ACME Order's
+	// status to valid. This is not an expected behaviour from an ACME server
+	// https://tools.ietf.org/html/rfc8555#section-7.1.6
+	// https://github.com/jetstack/cert-manager/issues/2868
+	case !anyChallengesFailed(challenges) && allChallengesFinal(challenges) && acmeOrder.Status == acmeapi.StatusPending:
+		log.V(logf.InfoLevel).Info("All challenges in a final state, waiting for ACME server to update the status of the order...")
+		// This is probably not needed as at this point the Order's status
+		// should already be Pending, but set it anyway to be explicit.
+		c.setOrderState(&o.Status, string(cmacme.Pending))
+		key, err := cache.MetaNamespaceKeyFunc(o)
+		if err != nil {
+			log.V(logf.InfoLevel).Info("Internal error: failed to construct key for pending Order")
```

Comment:
I'd be inclined to change this to a `log.Error` so that anyone who does encounter it will be likely to see it in their logs.

Reply:
Thanks 👍🏼